app/services/service_container.rb

```diff
@@ -0,0 +1,43 @@
+require "dry/container"
+
+class ServiceContainer
+  def initialize
+    @services = Dry::Container.new
```

Comment:
Why `Dry::Container` over simply defining a bunch of class methods and injecting them in there? E.g: 
```
def self.my_service
  ServiceClass.new(dependency: my_dependency)
end
```

Reply:
This route gets a lot harder to maintain as the services increase, with `Dry::Container` we can also put some of the service registration inside other folders to help split the file (e.g. The courses)